### ISSUE TITLE
QJsonArray's iterator doesn't return references, so acknowledge this in the code

### ISSUE THREAD

**Issue by iphydf** [qTox/qTox#6038]
src/net/bootstrapnodeupdater.cpp:186 says:
```
    for (const auto& node : nodes) {
```
which implies that we're taking a ref-to-const-QJsonValue, while actually we're making a temporary copy of a QJsonValueRef and then binding a const-ref to that. Instead, write:
```
    for (const QJsonValueRef node : nodes) {
```
to make it explicit that the copy is happening (it is, regardless of the `&` above), but also that it's of a lightweight "Ref" object.

**Comment by enkeyz** [comment]
Would like to work on this one.

**Comment by enkeyz** [comment]
Looks like it's fixed.

**Comment by anthonybilinski** [comment]
Confirmed, fixed in e71225268f91fe57afcd14d92a3a5307362e9aa7. Thanks.
